lib/sufia/export/actor.rb

```diff
@@ -0,0 +1,98 @@
+module Sufia
+  module Export
+    # Convert a GenericFile including metadata, permissions and version metadata into a PORO
+    # so that the metadata can be exported in json format using to_json
+    #
+    class Actor
+      attr_reader :converter_registry, :limit, :ids
+      attr_accessor :conversion_count
+
+      # initialize the class with the default registry
+      def initialize
+        @converter_registry = default_registry
+      end
+
+      # register a converter for a new class or overwrite the converter for a default class
+      #
+      # @param [Class] model_class     The ActiveFedora model class to be converter to json
+      # @param [Class] converter_class The class that will convert from ActiveFedora to json
+      def register_converter(model_class, converter_class)
+        raise(RegistryError, "Model (#{model_class.name}) for conversion must be an ActiveFedora::Base") unless model_class.ancestors.include?(ActiveFedora::Base)
+        raise(RegistryError, "Converter (#{converter_class.name}) for conversion must be an Sufia::Export::Converter") unless converter_class.ancestors.include?(Sufia::Export::Converter)
+        converter_registry[model_class] = converter_class
+      end
+
+      # Convert the classes using the registered converters from ActiveFedora to json files
+      #
+      # @param [Array] model_class_list list of classes to be converter
+      # @param [Hash]  opts
+      # @option opts [Number] :limit Limits the number of conversion done (defaults to -1 or all)
+      # @option opts [Array]  :ids   List of ids to be converted.  Can be from any model
+      def call(model_class_list = converter_registry.keys, opts = {})
+        @conversion_count = 0
+        validate_class_list(model_class_list)
+        parse_options(opts)
+        export_models(model_class_list)
+      end
+
+      private
+
+        def parse_options(opts)
+          @limit = opts[:limit] || -1
+          @ids = opts[:ids]
+        end
+
+        def model_scope(model_class)
+          scope = model_class.all
+          scope  = scope.where(id: ids) unless ids.blank?
```

Comment:
@jcoyne becuase that would error if the ID is not from that particular class. We could change the ids to take a hash with like ids: {GenericFile: ['abc123', '111222'], Collection: ['ccc111']} and know which class to expect the ids from, but the weher seemed simpler.